lib/deep-equal.js

```diff
@@ -18,176 +18,204 @@ var getTime = Date.prototype.getTime;
+        var actualType = typeof actualObj;
+        var expectationType = typeof expectationObj;
 
         // == null also matches undefined
         if (
-            obj1 === obj2 ||
-            isNaN(obj1) ||
-            isNaN(obj2) ||
-            obj1 == null ||
-            obj2 == null ||
-            type1 !== "object" ||
-            type2 !== "object"
+            actualObj === expectationObj ||
+            isNaN(actualObj) ||
+            isNaN(expectationObj) ||
+            actualObj == null ||
+            expectationObj == null ||
+            actualType !== "object" ||
+            expectationType !== "object"
         ) {
-            return identical(obj1, obj2);
+            return identical(actualObj, expectationObj);
         }
 
         // Elements are only equal if identical(expected, actual)
-        if (isElement(obj1) || isElement(obj2)) {
+        if (isElement(actualObj) || isElement(expectationObj)) {
             return false;
         }
 
-        var isDate1 = isDate(obj1);
-        var isDate2 = isDate(obj2);
-        if (isDate1 || isDate2) {
+        var isActualDate = isDate(actualObj);
+        var isExpectationDate = isDate(expectationObj);
+        if (isActualDate || isExpectationDate) {
             if (
-                !isDate1 ||
-                !isDate2 ||
-                getTime.call(obj1) !== getTime.call(obj2)
+                !isActualDate ||
+                !isExpectationDate ||
+                getTime.call(actualObj) !== getTime.call(expectationObj)
             ) {
                 return false;
             }
         }
 
-        if (obj1 instanceof RegExp && obj2 instanceof RegExp) {
-            if (valueToString(obj1) !== valueToString(obj2)) {
+        if (actualObj instanceof RegExp && expectationObj instanceof RegExp) {
+            if (valueToString(actualObj) !== valueToString(expectationObj)) {
                 return false;
             }
         }
 
-        if (obj1 instanceof Error && obj2 instanceof Error) {
-            return obj1 === obj2;
+        if (actualObj instanceof Error && expectationObj instanceof Error) {
+            return actualObj === expectationObj;
         }
 
-        var class1 = getClass(obj1);
-        var class2 = getClass(obj2);
-        var keys1 = keys(obj1);
-        var keys2 = keys(obj2);
-        var name1 = getClassName(obj1);
-        var name2 = getClassName(obj2);
-
-        if (isArguments(obj1) || isArguments(obj2)) {
-            if (obj1.length !== obj2.length) {
+        var actualClass = getClass(actualObj);
+        var expectationClass = getClass(expectationObj);
+        var actualKeys = keys(actualObj);
+        var expectationKeys = keys(expectationObj);
+        var actualName = getClassName(actualObj);
+        var expectationName = getClassName(expectationObj);
+        var expectationSymbols =
+            typeof Object.getOwnPropertySymbols === "function"
+                ? getOwnPropertySymbols(expectationObj)
```

Comment:
Is it possible that this line was supposed to read `? Object.getOwnPropertySymbols(expectationObj)`?

I just tried updating my project from sinon 7.2.7 to sinon 7.3.0, a change that includes this PR, and I get a lot of errors like this :
```
'Unhandled promise rejection', TypeError: undefined is not a constructor (evaluating 'getOwnPropertySymbols(expectationObj)')
deepEqual@http://localhost:9876/absolute/Users/angelika/myproject/node_modules/sinon/pkg/sinon.js?911775df668393e499a0995e2789094c48727cc3:4395:40
deepEqualCyclic@http://localhost:9876/absolute/Users/angelika/myproject/node_modules/sinon/pkg/sinon.js?911775df668393e499a0995e2789094c48727cc3:4493:7
...
```

They all go away when I change the line as I said.

Reply:
Interesting. Actually, the line obove should also just refer to `getOwnPropertySymbols` because we're "caching" the function at the top of the file. I would guess that your environment does not actually have symbols support, but it gets shimmed after Sinon is loaded.

Reply:
That is very probable, yes. Changing the above line to refer to just `getOwnPropertySymbols` without `Object.` also gets rid of my errors.

Reply:
Would you care to send a pull request to fix it?

Reply:
Here it is: https://github.com/sinonjs/samsam/pull/66